src/main/java/org/wmn4j/io/musicxml/MusicXmlWriterDom.java

```diff
@@ -766,6 +771,52 @@ private Element createPitchElement(Pitch pitch) {
 		return pitchElement;
 	}
 
+	private Optional<Element> createNotationsElement(Note note) {
+		final Element notationsElement = this.doc.createElement(MusicXmlTags.NOTATIONS);
+
+		// Articulations & fermata
+		if (note.hasArticulations()) {
```

Comment:
In the case there are no articulations, the creation of notationsElement could be avoided by moving it inside this if, or by returning empty at the beginning of the method. The only reason why in the attribute element creation the element is always created is that whether the attributes element is needed is more complicated to check. Generally it’s good to avoid creating unnecessary objects, at least when it’s easy to do.

Reply:
I was thinking that in the future this method will probably also include writing slurs and possibly dynamics, and similarly to the attribute element the checking might soon become complicated. Right now changing this is easy of course so if you think it's worth it I'll do it :stuck_out_tongue:

Reply:
I did not think of the extending of this method, so this is fine as it is :)